lightning/src/ln/channelmanager.rs

```diff
@@ -3253,7 +3291,18 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 				!htlcs.is_empty() // Only retain this entry if htlcs has at least one entry.
 			});
 		}
-		for failure in failed_channels.drain(..) {
+		for mut failure in failed_channels.drain(..) {
+			// It looks like our counterparty went on-chain. We cannot broadcast our latest local
+			// state via monitor update (as Channel::force_shutdown tries to make us do) as we may
+			// still be in initialization, so we track the update internally and handle it when the
+			// user next calls timer_chan_freshness_every_min, guaranteeing we're running normally.
```

Comment:
I'm seeing a pattern here possibly related to #800 where `ChannelManager`'s functionality is limited by the mode it's in. Do you think a refactoring is necessary to enforce this rather than relying on comments? Maybe that issue could be repurposed to be broader.

Reply:
Hmm, that's definitely a good point, I think we'd really want to figure out exactly what the API looks like in #800, though I have a feeling a lot of it is doing to end up being deep in the Channel state machine more than limits on the external API (we can always shove outbound things into channel holding cells anyway).